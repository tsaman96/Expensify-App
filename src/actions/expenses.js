import uuid from 'uuid';
import dataBase from '../firebase/firebase'
export const addExpense = ({
  description = '',
  note = '',
  amount = 0,
  createAt = 0
}={}) => ({
  type: 'ADD_EXPENSE',
  expense:{
  id: uuid(),
  description,
  note,
  amount,
  createAt
}
});

// export const startAddExpense = (expenseData = {}) => {
//   return (dispatch) => {
//     const { description = '',
//       note = '',
//       amount = 0,
//       createAt = 0
//     } = expenseData;
//     const expense = {description, note, amount, createAt};
//     dataBase.ref('expenses').push(expense).then((ref) => {
//       dispatch(addExpense({
//         id: ref.key,
//         ...expense
//       }))
//     }) ;
//   };
// };
//Remove Expense
export const removeExpense = ({ id } = {}) => ({
  type: 'REMOVE_EXPENSE',
  id
});
//Edit Expense
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates
})