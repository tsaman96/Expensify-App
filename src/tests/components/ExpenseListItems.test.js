import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItems from '../../components/ExpenseListItems';
import expenses from '../fixtures/expnses';

test('should render expense list item', () => {
  const wrapper = shallow(<ExpenseListItems {...expenses[2]}/>);
  expect(wrapper).toMatchSnapshot();
})