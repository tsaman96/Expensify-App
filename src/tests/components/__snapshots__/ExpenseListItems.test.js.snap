// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseListItems
    amount={1950000}
    createAt={345600000}
    description="credit card"
    id={3}
    note=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/edit/3"
>
          <h3>
                    credit card
          </h3>
</Link>,
        <p>
          1950000
           - 
          345600000
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h3>
            credit card
</h3>,
          "replace": false,
          "to": "/edit/3",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "credit card",
          },
          "ref": null,
          "rendered": "credit card",
          "type": "h3",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1950000,
            " - ",
            345600000,
          ],
        },
        "ref": null,
        "rendered": Array [
          1950000,
          " - ",
          345600000,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/edit/3"
>
            <h3>
                        credit card
            </h3>
</Link>,
          <p>
            1950000
             - 
            345600000
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <h3>
              credit card
</h3>,
            "replace": false,
            "to": "/edit/3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "credit card",
            },
            "ref": null,
            "rendered": "credit card",
            "type": "h3",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1950000,
              " - ",
              345600000,
            ],
          },
          "ref": null,
          "rendered": Array [
            1950000,
            " - ",
            345600000,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
